### DISCUSSION TITLE
Notification emails from Actions require SMTPUTF8

### DISCUSSION THREAD

**Discussion by ncfavier** [community/community#30063] [Actions]
I have recently stopped receiving email notifications for failed workflow runs (last one I got was on July 19), while other notifications are fine. I am running my own mail server with Postfix and Dovecot. Looking at the Postfix logs reveals this:

```
postfix/smtpd[2556826]: connect from out-21.smtp.github.com[192.30.252.204]
postfix/smtpd[2556826]: Anonymous TLS connection established from out-21.smtp.github.com[192.30.252.204]: TLSv1.2 with cipher ECDHE-ECDSA-AES256-GCM-SHA384 (256/256 bits)
policyd-spf[2556834]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=192.30.252.204; helo=smtp.github.com; envelope-from=notifications@github.com; receiver=<UNKNOWN>
postfix/smtpd[2556826]: 124259A2C11: client=out-21.smtp.github.com[192.30.252.204]
postfix/cleanup[2556835]: 124259A2C11: message-id=<ncfavier/1lab/check-suites/CS_kwDOHNlbUs8AAAAB1ho5Rg/1660901110@github.com>
postfix/qmgr[2261812]: 124259A2C11: from=<notifications@github.com>, size=37990, nrcpt=1 (queue active)
postfix/smtpd[2556826]: disconnect from out-21.smtp.github.com[192.30.252.204] ehlo=2 starttls=1 mail=1 rcpt=1 data=1 quit=1 commands=7
postfix/lmtp[2556836]: 124259A2C11: to=<n@monade.li>, relay=monade.li[/run/dovecot2/dovecot-lmtp], delay=3.3, delays=3.3/0.01/0.01/0, dsn=5.6.7, status=bounced (SMTPUTF8 is required, but was not offered by host monade.li[/run/dovecot2/dovecot-lmtp])
postfix/cleanup[2556835]: 4A8229A2F8E: message-id=<20220819092525.4A8229A2F8E@monade.li>
postfix/qmgr[2261812]: 4A8229A2F8E: from=<>, size=41165, nrcpt=1 (queue active)
postfix/bounce[2556838]: 124259A2C11: sender non-delivery notification: 4A8229A2F8E
postfix/qmgr[2261812]: 124259A2C11: removed
postfix/smtp[2556839]: 4A8229A2F8E: to=<notifications@github.com>, relay=aspmx.l.google.com[2a00:1450:400c:c06::1a]:25, delay=0.65, delays=0/0.02/0.17/0.46, dsn=2.0.0, status=sent (250 2.0.0 OK  1660901125 19-20020a05600c025300b0039c4ba160a8si3510936wmj.22 - gsmtp)
postfix/qmgr[2261812]: 4A8229A2F8E: removed
```

The GitHub SMTP server requires SMTPUTF8, but this is not supported by Dovecot (and it seems like there aren't any plans to add support) so Postfix bounces the email.

If I disable SMTPUTF8 in Postfix then the server disconnects immediately:

```
postfix/smtpd[2572306]: connect from out-18.smtp.github.com[192.30.252.201]
postfix/smtpd[2572306]: disconnect from out-18.smtp.github.com[192.30.252.201] ehlo=1 quit=1 commands=2
```

Is there any reason that SMTPUTF8 is now required for those emails?

Additional info: my name contains a non-ASCII character, and the `From:` header of those emails contains my name, which could explain why Postfix refuses to deliver the email without SMTPUTF8 support from Dovecot. (But other notification emails with non-ASCII characters in their `From:` are delivered just fine, and the `From:` is encoded with a `=?UTF-8?` marker.)

**Comment by vslavik** [comment]
Thanks for documenting the bug publicly. 

I'll add that your name is definitely what triggers it. FastMail (and I'd imagine a lot of other providers too) is also affected by this and the workaround was to edit my name down to ASCII-only in GitHub profile.

**Comment by ncfavier** [comment]
Yeah, I'm definitely not doing that :)

I should mention that the support ticket I opened in August just got the following update today:

> Our engineering team have taken it up and they'll be working to fix it in future releases.
>
> I am unable to give you a definite timeline of when the fix would be implemented but you can keep an eye on our [official GitHub public product roadmap](https://github.com/github/roadmap), our [Changelog](https://github.blog/changelog/) and our [GitHub Blog](http://github.blog/) for future updates.

**Comment by ncfavier** [comment]
This seems to be fixed now!
